rust/lance-encoding/src/encodings/logical/primitive.rs

```diff
@@ -132,12 +135,19 @@ impl DecodeArrayTask for PrimitiveFieldDecodeTask {
     fn decode(self: Box<Self>) -> Result<ArrayRef> {
         // There are two buffers, the validity buffer and the values buffer
         // We start by assuming the validity buffer will not be required
-        let mut capacities = [(0, false), (0, true)];
+        let mut capacities = vec![(0, false); self.physical_decoder.num_buffers() as usize];
```

Comment:
Comment above probable needs to be updated, right?

Reply:
Updated